modules/ossm-threescale-webassembly-module-examples-for-credentials-use-cases.adoc

```diff
@@ -0,0 +1,236 @@
+// Module included in the following assembly:
+//
+// service_mesh/v2x/ossm-threescale-webassembly-module.adoc
+
+[id="ossm-threescale-webassembly-module-examples-for-credentials-use-cases_{context}"]
+= The 3scale WebAssembly module examples for `credentials` use cases
+
+You will spend most of your time applying configuration steps to obtain credentials in the requests to your services.
+
+The following are `credentials` example, which you can modify to adapt to specific use cases.
+
+You can combine them all, although when you specify multiple source objects with their own `lookup queries`, they are evaluated in order until one of them successfully resolves.
+
+[id="api-key-in-query-string-parameters_{context}"]
+== API key (`user_key`) in query string parameters
+This looks up a `user_key` in a query string parameter or header of the same name:
+
+[source,yaml]
+----
+credentials:
+  user_key:
+    - query_string:
+        keys:
+          - user_key
+    - header:
+        keys:
+          - user_key
+----
+
+[id="application-id-and-key_{context}"]
+== Application ID and `key`
+The following example looks up a `app_key` and `app_id` in a query or headers.
+
+[source,yaml]
+----
+credentials:
+  app_id:
+    - header:
+        keys:
+          - app_id
+    - query_string:
+        keys:
+          - app_id
+  app_key:
+    - header:
+        keys:
+          - app_key
+    - query_string:
+        keys:
+          - app_key
+----
+
+[id="authorization-header_{context}"]
+== Authorization header
+A request includes an `app_id` and `key` in an `authorization` header. If there is at least one or two values outputted at the end, then you can assign the `app_key`.
+
+The resolution here will assign the `application key` if there is one or two output at the end.
+
+The `authorization` header specifies a value with the type of authorization and its value is encoded as `Base64`. This means you can split the value by a space character, take the second output and then split it again using a colon (:) as the separator. For example, if you use this format: `app_id:app_key`, the header will look like this for `credential`:
+
+`aladdin:opensesame`:  Authorization: Basic YWxhZGRpbjpvcGVuc2VzYW1l
```

Comment:
Maybe this is already fine, but wondering if putting it in a code block helps? In case something like this is more accurate:

```
aladdin:opensesame:  Authorization: Basic YWxhZGRpbjpvcGVuc2VzYW1l
```

Was just curious about the current mix of backtick markup with plain text.